### ISSUE TITLE
[C+ Checklist Needs Completion] [$1000]  Delete option is shown for threaded chat but not parent in deleted workspace

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#23604]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to announce room
2. Send a message
3. Delete the workspace
4. Notice that the delete option doesn't show when hovering over the message
5. Click on reply in thread of the message
6. Hover over the thread first chat
## Expected Result: 
App should removed the red dot from LHN as soon as we dismiss the error
## Actual Result: 
App does not dismiss red dot from LHN when we delete message in archived room from reply in thread and dismiss the triggered error

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.45-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/142a65f3-5487-4de4-ad9d-116e396c8de2

https://github.com/Expensify/App/assets/43996225/50faf263-35f5-4697-816b-55090f4ce443

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1690268782561039

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d9f5ac42d3564a9d</li>
        <li>Upwork Job ID: 1684533698073559040</li>
        <li>Last Price Increase: 2023-08-03</li>
<li>Automatic offers: </li>
<ul>
<li>s77rt | Contributor | 26038579</li>
<li>dhanashree-sawant | Reporter | 26038580</li>
<li>dukenv0307 | Contributor | 26142842</li>
</ul></li>
    </ul>
</details>

**Comment by greg-schroeder** [comment]
Reviewing

**Comment by greg-schroeder** [comment]
@0xmiroslav mind taking a look at @dukenv0307's proposal here? https://github.com/Expensify/App/issues/23604#issuecomment-1650861123

**Comment by LM0305** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
In archived workspace room, error indicator(red dot) from LHN is not removed when we dismiss error
### What is the root cause of that problem?
The Report screen page doesn't re-render when close report action alert description. this is the reason that doesn't update LHN.
### What changes do you think we should make in order to solve the problem?
Need to add re-render action script in ReportActions.js
https://github.com/Expensify/App/blob/714950e15182c595e0229734972c905ba2278136/src/libs/actions/ReportActions.js#L10-L31

### What alternative solutions did you explore? (Optional)

### Result

https://github.com/Expensify/App/assets/141049840/770c01ea-5522-42a1-be28-3422e1027c23

**Comment by LM0305** [comment]
Contributor details
Your Expensify account email: muhammadafzal0305@outlook.com
Upwork Profile Link: https://www.upwork.com/freelancers/~018ef9deb21c95d972

**Comment by LM0305** [comment]
Contributor details
Your Expensify account email:muhammadafzal0305@outlook.com
Upwork Profile Link: https://www.upwork.com/freelancers/~018ef9deb21c95d972

**Comment by StanislavLavrenchuk1** [comment]
@LM0305's proposal looks good

**Comment by StanislavLavrenchuk1** [comment]
Contributor details
Your Expensify account email: delorash091ash@hotmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~011bbf31f557541361

**Comment by greg-schroeder** [comment]
Just waiting on proposal review from @0xmiroslav :)

**Comment by greg-schroeder** [comment]
@0xmiroslav bump on proposal review

**Comment by s77rt** [comment]
Taking this as C+

**Comment by greg-schroeder** [comment]
Thank you @s77rt!

**Comment by s77rt** [comment]
Is this still reproducible? It seems I can delete the message without getting any backend error

**Comment by s77rt** [comment]
@dukenv0307 Actually I can reproduce now, I had to create a new WS.

**Comment by s77rt** [comment]
@dukenv0307 Thanks for the proposal. Your RCA makes sense. I don't think we should subscribe for report actions as it may lead to unnecessary renders. For this case I think it's better to go with the alternative solution which is to hide the delete option for the thread's first message if the parent is an archived room (because the option is hidden in the parent). We should do the same for the edit and flag options.

:ribbon: :eyes: :ribbon: C+ reviewed
[Link to proposal](https://github.com/Expensify/App/issues/23604#issuecomment-1650861123)

**Comment by s77rt** [comment]
@LM0305 Thanks for the proposal. Your RCA makes sense but opening the report every time we clear the errors seems an overkill and a workaround.

**Comment by s77rt** [comment]
@dukenv0307 The whole error shouldn't be there. The delete, edit and flag options should not be displayed because they are not displayed in the parent. Your alternative solution seems the best to do here.

**Comment by srikarparsi** [comment]
Hey! Sorry for the delayed response, was OOO friday and monday. I'm also not able to reproduce the [posted issue](https://github.com/Expensify/App/issues/23604#issue-1821448612) on dev. @dukenv0307, do you think you could write down reproducible test steps for the new issue you're [describing](https://github.com/Expensify/App/issues/23604#issuecomment-1672552918)? I also think we should create a new issue for this since it doesn't seem related to the first issue posted.

**Comment by srikarparsi** [comment]
Cool, thanks! Yup, I'll just update the title and test steps. The proposal to hide delete, edit and flag to match the parent also makes sense.

**Comment by greg-schroeder** [comment]
I'm not sure what's up with the automation... maybe it's delayed for some reason? the date would be 9/4 if it doesn't work

**Comment by greg-schroeder** [comment]
@dukenv0307 payout date would have been 9/4 but that was a holiday. So it's today! I'll take care of it

**Comment by greg-schroeder** [comment]
Issue Participants:

Issue reported by: @dhanashree-sawant
Contributor: @dukenv0307
C+: @s77rt 

Was this issue merged in time to be eligible for the speed bonus? ✅ 
Were their any regressions? ❌ 

Payment summary:

Reporter: $250
Contributor: $1,500
C+: $1,500

**Comment by s77rt** [comment]
- The PR that introduced the bug has been identified: This existed since the implementation of threads. This is basically a missing feature - context menu didn't support threads well.
- The offending PR has been commented on: n/a
- A discussion in #expensify-bugs has been started: Not needed. This is more like a feature request
- Determine if we should create a regression test for this bug: No

**Comment by s77rt** [comment]
@greg-schroeder Completed ^

**Comment by greg-schroeder** [comment]
Thanks @s77rt!
